delfin/drivers/ibm/storwize_svc/ssh_handler.py

```diff
@@ -0,0 +1,409 @@
+                    'wwn': str(map.get('vdisk_UID')),
+                    'type': volume_type,
+                    'total_capacity': int(total_capacity),
+                    'used_capacity': int(used_capacity),
+                    'free_capacity': int(free_capacity),
+                    'compressed': compressed,
+                    'deduplicated': deduplicated
+                }
+                volume_list.append(v)
+
+            return volume_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get storage volume: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get storage volume: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def judge_alert_time(self, map, query_para):
+        if len(map) <= 1:
+            return False
+        if query_para is None and len(map) > 1:
+            return True
+        occur_time = int(map.get('last_timestamp_epoch')) * 1000
+        if query_para.get('begin_time') and query_para.get('end_time'):
+            if occur_time >= int(query_para.get('begin_time')) and \
+                    occur_time <= int(query_para.get('end_time')):
+                return True
+        if query_para.get('begin_time'):
+            if occur_time >= int(query_para.get('begin_time')):
+                return True
+        if query_para.get('end_time'):
+            if occur_time <= int(query_para.get('end_time')):
+                return True
+        return False
+
+    def list_alerts(self, context, query_para):
+        try:
+            alert_list = []
+            alert_info = self.handle_detail_ssh(
+                context, self.eventlog_command_without_filter, 'alert')
+            alert_res = alert_info.split('\n')
+            for i in range(1, len(alert_res)):
+                if alert_res[i] is None or alert_res[i] == '':
+                    continue
+                alert_str = ' '.join(alert_res[i].split())
+                strinfo = alert_str.split(' ', 9)
+                detail_command = \
+                    SSHHandler.alert_detail_command + strinfo[0]
+                deltail_info = self.handle_detail_ssh(
+                    context, detail_command, 'alertdetail')
+                map = {}
+                self.handle_detail(deltail_info, map, split=' ')
+                if self.judge_alert_time(map, query_para) is False:
+                    continue
+                time_stamp = int(map.get('last_timestamp_epoch')) * 1000
+                alert_name = map.get('event_id_text') and map.get(
+                    'event_id_text') or ''
+                event_id = map.get('event_id')
+                location = map.get('object_name') and map.get('object_name') \
+                    or ''
+                resource_type = map.get('object_type') and map.get(
```

Comment:
what's the priority of `and` & `or`, should this "A and B or C" be "(A and B) or C" or "A and (B or C)".
For example, if A is false, B is true, C is true, "(A and B) or C" is true, "A and (B or C)" is false, what do you suppose "A and B or C" to return?